File: ios/Classes/MxCorePlugin.m
#import "MxCorePlugin.h"
#import <mx_core/mx_core-Swift.h>

@implementation MxCorePlugin
+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
  [SwiftMxCorePlugin registerWithRegistrar:registrar];
}
@end

File: ios/Classes/MxCorePlugin.swift
import Flutter
import UIKit

public class MxCorePlugin: NSObject, FlutterPlugin {
  public static func register(with registrar: FlutterPluginRegistrar) {
    let channel = FlutterMethodChannel(name: "mx_core", binaryMessenger: registrar.messenger())
    let instance = MxCorePlugin()
    registrar.addMethodCallDelegate(instance, channel: channel)
  }

  public func handle(_ call: FlutterMethodCall, result: @escaping FlutterResult) {
    switch call.method {
    case "getPlatformVersion":
      result("iOS " + UIDevice.current.systemVersion)
    default:
      result(FlutterMethodNotImplemented)
    }
  }
}
